### PR TITLE
Casting (0,1) values to bit in the select list to avoid casting excep…

### PR DESCRIPTION
**Pull request by malikhassan85**
Change: Casting (0,1) values to bit in the select list to avoid casting exception when casting these values to boolean at code level.

Please note that for the integration test for this scenario, EFCore team covered this scenario under a test called "**Correlated_collection_with_complex_order_by_funcletized_to_constant_bool**" in "**EFCore.SqlServer.FunctionalTests**" project, however, this test has a compilation condition and it's not part from the final package., that's why I couldn't override it in you functional test project. So please feel free to point me on where should I add a test to cover this scenario, or feel free to add it by yourself if you want.

### CODE REVIEW

File: src/Provider35/EntityFramework.SqlServerCompact35.csproj [hunk 1]
```diff
@@ -98,6 +104,7 @@
     </Reference>
     <Reference Include="System.ComponentModel.Composition" />
     <Reference Include="System.ComponentModel.DataAnnotations" />
+    <Reference Include="System.Configuration" />
```
Comment (ErikEJ) [inline src/Provider35/EntityFramework.SqlServerCompact35.csproj]: Should probably be removed

File: src/Provider35/EntityFramework.SqlServerCompact35.csproj [hunk 2]
```diff
@@ -368,6 +375,7 @@
     </Compile>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (ErikEJ) [inline src/Provider35/EntityFramework.SqlServerCompact35.csproj]: Please remove

File: src/Provider35/app.config [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ErikEJ) [inline src/Provider35/app.config]: Delete

File: src/Provider40/EntityFramework.SqlServerCompact40.csproj [hunk 4]
```diff
@@ -211,8 +218,11 @@
     <Compile Include="Utilities\SqlCeHelper.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (ErikEJ) [inline src/Provider40/EntityFramework.SqlServerCompact40.csproj]: Pls remove

File: src/Provider40/app.config [hunk 5]
```diff
@@ -0,0 +1,35 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ErikEJ) [inline src/Provider40/app.config]: Delete

**Comment by ErikEJ** [comment]
Looks good, pls remove the app.config from the 2 product projects

**Comment by ErikEJ** [comment]
Look good, I will merge when you other PR is in a has been released

**Comment by malikhassan85** [comment]
Hi Erik,

I just wonder if you have any roughly date for when the new version which includes this PR will get released? from my understanding this version should depend on EFCore 2.1.2.

**Comment by ErikEJ** [comment]
Have you tested the Myget package?

**Comment by ErikEJ** [comment]
I think you need to resubmit the PR, it looks like you reverted it along the way, and it was never merged - sorry for not realizing..

**Pull request merged**